File: frontend/src/utils/workingDays.ts
// Simple client-side day-count calculator used ONLY as a labeled "preview"
// while the user fills the form — always defer to the server-computed value
// on submit. Counts every calendar day inclusive of both endpoints,
// including weekends.
export function previewWeekdayCount(startDate: string, endDate: string): number {
  if (!startDate || !endDate) return 0;
  const start = new Date(startDate);
  const end = new Date(endDate);
  if (Number.isNaN(start.getTime()) || Number.isNaN(end.getTime()) || end < start) return 0;

  const msPerDay = 24 * 60 * 60 * 1000;
  return Math.round((end.getTime() - start.getTime()) / msPerDay) + 1;
}

File: backend/apps/leave/workingdays.py
"""
Server-authoritative leave-day calculator.

Counts every calendar day in the range, inclusive of both endpoints —
weekends are included in the count. This is the single source of truth
used both when an application is created/updated and via the standalone
preview endpoint the frontend calls for live totals.
"""


def calculate_working_days(start_date, end_date):
    """
    Count calendar days (inclusive of both endpoints) between start_date and
    end_date, including weekends. Returns 0 if end_date < start_date.
    """
    if end_date < start_date:
        return 0
    return (end_date - start_date).days + 1
